dot15d4/src/phy/driver.rs

```diff
@@ -0,0 +1,127 @@
+use core::future::Future;
+use core::task::Context;
+use core::task::Poll;
+
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Error {
+    CCAFailed,
+    ACKFailed,
+    InvalidStructure,
+    RadioError,
+}
+
+/// Should be given as an argument to the task that will run the network protocol.
+/// This trait allows to abstract over channels in async executors.
+pub trait Driver {
+    /// Waits until there is something to be transmitted
+    fn transmit(&self) -> impl Future<Output = PacketBuffer>;
+    /// Hold until the buffer is received successfully
+    fn received(&self, buffer: PacketBuffer) -> impl Future<Output = ()>;
+    /// Hold until the buffer is received successfully
+    fn error(&self, error: Error) -> impl Future<Output = ()>;
+}
+
+/// A buffer that is used to store 1 frame.
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, Clone, PartialEq)]
+pub struct PacketBuffer {
```

Comment:
Rename this to `FrameBuffer` as we are using frames instead of packets. Packets are higher up in the network stack.